driver-core/src/test/functional/com/mongodb/ClusterFixture.java

```diff
@@ -201,20 +189,27 @@ public static synchronized ConnectionString getConnectionString() {
         return connectionString;
     }
 
+    public static ReadWriteBinding getBinding(final Cluster cluster) {
```

Comment:
If a test passes in its own cluster, don't use the cached binding from the default cluster.